qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/HybridSearchIT.java

```diff
@@ -0,0 +1,163 @@
+import static org.opensearch.neuralsearch.TestUtils.DEFAULT_NORMALIZATION_METHOD;
+import static org.opensearch.neuralsearch.TestUtils.DEFAULT_COMBINATION_METHOD;
+import org.opensearch.neuralsearch.query.HybridQueryBuilder;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+public class HybridSearchIT extends AbstractRestartUpgradeRestTestCase {
+    private static final String PIPELINE_NAME = "nlp-hybrid-pipeline";
+    private static final String PIPELINE1_NAME = "nlp-hybrid-1-pipeline";
+    private static final String SEARCH_PIPELINE_NAME = "nlp-search-pipeline";
+    private static final String SEARCH_PIPELINE1_NAME = "nlp-search-1-pipeline";
+    private static final String TEST_FIELD = "passage_text";
+    private static final String TEXT_1 = "Hello world";
+    private static final String TEXT_2 = "Hi planet";
+    private static final String TEXT_3 = "Hi earth";
+    private static final String TEXT_4 = "Hi amazon";
+    private static final String TEXT_5 = "Hi mars";
+    private static final String TEXT_6 = "Hi opensearch";
+    private static final String QUERY = "Hi world";
+
+    // Test restart-upgrade normalization processor when index with multiple shards
+    // Create Text Embedding Processor, Ingestion Pipeline, add document and search pipeline with normalization processor
+    // Validate process , pipeline and document count in restart-upgrade scenario
+    public void testNormalizationProcessor_whenIndexWithMultipleShards_E2EFlow() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        if (isRunningAgainstOldCluster()) {
+            String modelId = uploadTextEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId, PIPELINE_NAME);
+            createIndexWithConfiguration(
+                getIndexNameForTest(),
+                Files.readString(Path.of(classLoader.getResource("processor/IndexMappingMultipleShard.json").toURI())),
+                PIPELINE_NAME
+            );
+            addDocument(getIndexNameForTest(), "0", TEST_FIELD, TEXT_1, null, null);
+            addDocument(getIndexNameForTest(), "1", TEST_FIELD, TEXT_2, null, null);
+            addDocument(getIndexNameForTest(), "2", TEST_FIELD, TEXT_3, null, null);
+            addDocument(getIndexNameForTest(), "3", TEST_FIELD, TEXT_4, null, null);
+            addDocument(getIndexNameForTest(), "4", TEST_FIELD, TEXT_5, null, null);
+            createSearchPipeline(
+                SEARCH_PIPELINE_NAME,
+                DEFAULT_NORMALIZATION_METHOD,
+                DEFAULT_COMBINATION_METHOD,
+                Map.of(PARAM_NAME_WEIGHTS, Arrays.toString(new float[] { 0.3f, 0.7f }))
+            );
+        } else {
+            Map<String, Object> pipeline = getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId = getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
```

Comment:
I think two lines above, 63-64, should be inside the getModel method, same applies to similar code in other methods. Search pipeline is not part of the hybrid search functionality, the only thing we need from pipeline is the modelId. Moving that to method should increase code reuse and improve readability of the test method.

Reply:
I don't see where this comment has been addressed. You've created a method that takes the pipeline name and uses some default processor name, you need similar that takes pipeline name and the processor name. Something like:

```
String modelId = getModelId(PIPELINE_NAME, TEXT_EMBEDDING_PROCESSOR)
```

Reply:
getModelId is in TestUtils and TestUtils does not inherit BaseNeuralSearchIT. Therefore I will not be able to call getIngestionPipeline method from there. TestUtils is just a class where we keep the constants and static methods with no operations.

Reply:
Currently, it does not take a pipeline name, rather it takes get ingestonPipeline response

Reply:
ok, got it. We could pass the function from one class to another via Functional interface, but that does not look better than current approach.